.github/workflows/latex-build.yml

```diff
@@ -0,0 +1,49 @@
+name: Build Beamer Presentation
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+
+      - name: Install minimal TeX Live packages
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y \
+            texlive-latex-base \
+            texlive-latex-recommended \
+            texlive-fonts-recommended \
+            texlive-latex-extra \
+            texlive-lang-czechslovak
+
+      - name: Clean previous build files
+        run: rm -f presentation/*.pdf presentation/*.aux presentation/*.log
+
+      - name: Compile Beamer Presentation
+        run: |
+          pdflatex -output-directory=presentation presentation/prezentace.tex
+          pdflatex -output-directory=presentation presentation/prezentace.tex
+
+      - name: Commit and push PDF file
+        run: |
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+          git add presentation/prezentace.pdf
+          git commit -m "Add generated Beamer PDF"
+          git push
+
+      - name: Upload PDF artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: beamer-presentation
+          path: presentation/prezentace.pdf
```

Comment:
Tady místo git push poprosím přes github akce vytvořit release a nahrát artefact do něj (pokud je to on tag).
Pokud je to on pull tak bych to dal možná jako comment do toho daného PR s odkazem to generované pdf